### ISSUE TITLE
Rework settings and update to IfcOpenShell v0.8

### ISSUE THREAD

**Issue by hlg** [opensourceBIM/IfcOpenShell-BIMserver-plugin#17]
There have been changes to the way haow IfcOpenShell handles settings from v0.8 on. The rework might also consider new settings such as  "permissive extrusions" (allows extrusions depths of zero or less). Parallel adjustments might need to be made on the geometry server side in IfcOpenShell itself.

See https://github.com/IfcOpenShell/IfcOpenShell/issues/1490#issuecomment-2306598309 for reference.

**Comment by hlg** [comment]
There have been changes already from [v0.6.0](https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.6.0/src/ifcgeom/IfcGeomIteratorSettings.h) to [v0.7.0](https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/ifcgeom_schema_agnostic/IfcGeomIteratorSettings.h).

In v0.6.0 we have setting IDs as follows:

~~~
SEW_SHELLS = 1 << 4,
DISABLE_OPENING_SUBTRACTIONS = 1 << 6,
APPLY_LAYERSETS = 1 << 13,
~~~

In v0.7.0 due to the [removal of the FASTER_BOOLEAN setting](https://github.com/IfcOpenShell/IfcOpenShell/commit/ff73f82798a8e744a0b7e2aaa353252df0a2934c), there is:

~~~
SEW_SHELLS = 1 << 4,
DISABLE_OPENING_SUBTRACTIONS = 1 << 5,
APPLY_LAYERSETS = 1 << 12,
~~~

Even though I don't know if this accounts for all 0.7.0 builds, to handle this change, we can only make a distinction on the basis of the version that the geom server announces during the HELLO handshake. It may be that some earlier 0.7.0 builds will fail with this, but the later ones should work. It is a pity that there is no semantic versioning on the minor number. Commit SHAs are not consecutive, we would have to enumerate the SHAs of the builds in order to distinguish the builds before and after the FASTER_BOOLEAN removal.

**Comment by aothms** [comment]
Yeah... sorry bout that. In 0.8 settings are identified by string so wouldn't affect us like that anymore in the future.
